### PR TITLE
Integrate email template in cli

### PR DESCRIPTION
**Pull request by rupal-bq**
### Description
- Integrated email template (for email body image, created a temporary png image for all format and deleted it before cli exists)

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/arguments.js [hunk 1]
```diff
@@ -56,9 +56,12 @@ export async function getCommandArguments() {
             .env(ENV_VAR.SMTP_USERNAME))
         .addOption(new Option('--smtppassword <password>', 'smtp password')
             .env(ENV_VAR.SMTP_PASSWORD))
-        .addOption(new Option('--subject <subject>', 'email Subject')
+        .addOption(new Option('--subject <subject>', 'email subject')
             .default(DEFAULT_EMAIL_SUBJECT)
             .env(ENV_VAR.EMAIL_SUBJECT))
+        .addOption(new Option('--note <subject>', 'email note')
```
Comment (joshuali925) [inline src/arguments.js:62]: does `email note` mean email body?

File: src/constants.js [hunk 2]
```diff
@@ -11,6 +11,7 @@ export const DEFAULT_WIDTH = '1680';
 export const DEFAULT_MIN_HEIGHT = '600';
 export const DEFAULT_FILENAME = 'opensearch-report';
 export const DEFAULT_EMAIL_SUBJECT = 'This is an email containing your opensearch dashboard report';
+export const DEFAULT_EMAIL_NOTE = 'Hi,<br>Here is the latest report!';
```
Comment (joshuali925) [inline src/constants.js:14]: will users know to use `<br>` instead of new line?

File: src/download-helpers.js [hunk 3]
```diff
@@ -132,12 +132,19 @@ export async function downloadReport(url, format, width, height, filename, authT
...
     const data = { timeCreated, dataUrl: buffer.toString('base64'), };
-
     await readStreamToFile(data.dataUrl, filename, format);
+
+    if (transport !== undefined) {
+      let emailTemplateImageBuffer = await page.screenshot({
+        fullPage: true,
+      });
+      const data = { timeCreated, dataUrl: emailTemplateImageBuffer.toString('base64'), };
+      await readStreamToFile(data.dataUrl, 'email_body.png', FORMAT.PNG);
```
Comment (joshuali925) [inline src/download-helpers.js:144]: email doesn't support csv?

**Review by joshuali925 — COMMENTED** [review]
some questions

**Comment by rupal-bq** [comment]
https://user-images.githubusercontent.com/56703525/214382895-d9555a52-c42f-44f0-9a6c-53cbcfbbacf2.mov

File: src/arguments.js (hunk 1 above)
Comment (rupal-bq) [reply src/arguments.js:62]: No, It's the note section in email body
image

File: src/download-helpers.js (hunk 3 above)
Comment (rupal-bq) [reply src/download-helpers.js:144]: CSV is attached with email and this image is added in the html body. Check this video and let me know if we should change something

https://user-images.githubusercontent.com/56703525/214384721-6bc77e02-9b01-47ce-b3fc-6bd53486ecfa.mov

File: src/constants.js (hunk 2 above)
Comment (rupal-bq) [reply src/constants.js:14]: not sure. should we add template is html somewhere in the docs?

File: src/download-helpers.js (hunk 3 above)
Comment (joshuali925) [reply src/download-helpers.js:144]: got it, looks good

File: src/arguments.js (hunk 1 above)
Comment (joshuali925) [reply src/arguments.js:62]: would users know what it means? i've not heard email note before

File: src/constants.js (hunk 2 above)
Comment (joshuali925) [reply src/constants.js:14]: what happens if new line is provided? if doesn't work as expected maybe programmatically convert text to html?

just a thought but does this need to support reading from files for easier multi-line editing? e.g. `--email-body-file ./email-body.txt`. or users are likely to use one-line notes

Comment (rupal-bq) [reply src/constants.js:14]: currently, if new line is provided it just gets ignored 
e.g. 'Hi,\nHere is the latest report!' will look like
image

We can surely add text to html logic and/or read from file. Not sure what is preferred though. will check

File: src/download-helpers.js [hunk 4]
```diff
@@ -132,12 +132,19 @@ export async function downloadReport(url, format, width, height, filename, authT
...
 
-    await browser.close();
-
     const timeCreated = time.valueOf();
     const data = { timeCreated, dataUrl: buffer.toString('base64'), };
-
     await readStreamToFile(data.dataUrl, filename, format);
+
+    if (transport !== undefined) {
+      let emailTemplateImageBuffer = await page.screenshot({
```
Comment (mengweieric) [inline src/download-helpers.js:140]: Minor: I didn't see value of emailTemplateImageBuffer being changed after, we could use const

File: src/email-helpers.js [hunk 5]
```diff
@@ -22,11 +22,12 @@ try {
   // Do not set AWS_SDK_LOAD_CONFIG if aws config file is missing.
 }
 
-export async function sendEmail(filename, format, sender, recipient, transport, smtphost, smtpport, smtpsecure, smtpusername, smtppassword, subject) {
+export async function sendEmail(filename, url, sender, recipient, transport, smtphost, smtpport, smtpsecure, smtpusername, smtppassword, subject, note) {
```
Comment (mengweieric) [inline src/email-helpers.js:25]: Minor: I feel the param list is a little bit long, we may consider to refactor it a little bit later.

**Review by mengweieric — APPROVED** [review]
Left minor comments

File: src/email-helpers.js (hunk 5 above)
Comment (rupal-bq) [reply src/email-helpers.js:25]: created issue #12 to track this

File: src/download-helpers.js (hunk 4 above)
Comment (rupal-bq) [reply src/download-helpers.js]: Thanks! used const.

File: src/arguments.js (hunk 1 above)
Comment (rupal-bq) [reply src/arguments.js]: does email body or message makes more sense? any other suggestion?

File: src/arguments.js [hunk 6]
```diff
@@ -174,7 +175,23 @@ function getOptions(options) {
...
+// Convert text to html
+function getHtml(text) {
+    text = (text || "");
+    return text
+        .replace(/&/g, "&amp;")
+        .replace(/</g, "&lt;")
+        .replace(/>/g, "&gt;")
+        .replace(/\t/g, "    ")
+        .replace(/ /g, "&#8203;&nbsp;&#8203;")
+        .replace(/\r\n|\r|\n/g, "<br />");
```
Comment (joshuali925) [inline src/arguments.js:196]: does this need `"` and `'`? (`&quot;` and `&apos;`)

**Review by joshuali925 — APPROVED** [review]

File: src/arguments.js (hunk 6 above)
Comment (rupal-bq) [reply src/arguments.js:196]: maybe not needed. tested with text file note 
image

**Pull request merged**